compiler/backend/src/org/jetbrains/kotlin/codegen/DelegatedPropertiesCodegenHelper.kt

```diff
@@ -65,13 +67,14 @@ class DelegatedPropertiesCodegenHelper(private val state: GenerationState) {
             this
 
     private fun isDelegatedPropertyMetadataRequiredForFunctionFromBinaries(calleeDescriptor: FunctionDescriptor): Boolean {
-        assert(calleeDescriptor is DescriptorWithContainerSource) {
+        require(calleeDescriptor is DescriptorWithContainerSource) {
             "Function descriptor from binaries expected: $calleeDescriptor"
         }
-
         val metadataParameterIndex = getMetadataParameterIndex(calleeDescriptor)
-        val methodNode = InlineCodegen.createSpecialInlineMethodNodeFromBinaries(calleeDescriptor, state)
-
+        val containerId = KotlinTypeMapper.getContainingClassesForDeserializedCallable(calleeDescriptor).implClassId
```

Comment:
Could you also add bytecode tests for optimization in `$default ` case and mangled ones: inline classes, suspend functions

Reply:
sorry, I didn't get what you mean

Reply:
There several bytecode tests covers this code https://github.com/JetBrains/kotlin/tree/b15e32936e22f9510c9a21ebe6f194c40cdd9620/compiler/testData/codegen/bytecodeText/optimizedDelegatedProperties
Could you add new ones for mentioned cases because mangled ($default, inline classes, suspend) ones are not covered here

Reply:
Sorry, I still don't get what exactly is supposed to have default parameters/inline classes/suspensions here. I think out of all that only inline classes can apply to `operator fun {get,set}Value`?

Reply:
Oh, indeed, it's works in opposite way than I thought, sorry for inconvenience